### DISCUSSION TITLE
How to load a document directly into the prompt or into context rather than a vector store

### DISCUSSION THREAD

**Discussion by vertgo** [FlowiseAI/Flowise#2427] [Q&A]
Hi, I want to upload a document and it goes into the context and skipping the step of upserting and querying the vector store (the documents will be small enough to fit in context). Any way to do so? They will be csv at first. If it works, I can try to expand to pdf.

**Comment by toi500** [comment]
Did you try the CSV agent?

https://toi500.gitbook.io/flowise-docs/integrations/langchain/agents/csv-agent

**Reply by njfio** [reply]
@toi500 I just wanted to thank you for all your effort in that doc repo! Flowise is pushing forward at a breakneck speed, and it's great to see how the community of users fills in as the gaps start to grow.

**Reply by toi500** [reply]
Happy to help, man.

I opened a pull request (https://github.com/FlowiseAI/FlowiseDocs/pull/70) and I'm still waiting to hear from @HenryHengZJ.

I think it's really important for users to know, at least, which nodes are available and what they do. We can then build on this foundation by asking the community to help with the technical details for each node.

**Reply by vertgo** [reply]
I did, but I didn't want the csv agent. I had my own prompts that I wanted to frame how to treat the data in a csv, and wanted to just drop the contents into a prompt template directly. I basically want a node to upload the file and output the text.

**Comment by BuildersRejected** [comment]
This is possible. I'm doing it with PlainText and should work with CSV.

Connect the CSV Document Loader to the Prompt Template and click the "Format Prompt Variables" button. There you will create a key/value pair. They key can be whatever you choose (i.e. "csv_data"). When entering the value, a dropdown will show an option for the CSV loader. `"{{csvFile_0.data.instance}}"`

Then your prompt template can include the key. Example:

> Use this data to answer questions: {csv_data}

![image](https://github.com/FlowiseAI/Flowise/assets/48058454/90405f87-c93c-4bb7-951e-5cda5dee5cce)
![image](https://github.com/FlowiseAI/Flowise/assets/48058454/bccc0122-857c-4f62-8d54-75a158cc22cd)

P.S. It IS also possible to send this CSV data through API along with a user's query.

**Comment by josh-at-straker** [comment]
I am also looking to do this, but the process really isn't all that clear. 

I want to poll one of my APIs, get the JSON output, and have the LLM use this data in it's context / prompt template. 
Whats the best option to do this with the current nodes we have avalible?
